### ISSUE TITLE
Lxqt-leave blank

### ISSUE THREAD

**Issue by ferronrsmith** [lxde/lxqt#1282]
lxqt-leave stop working (showing blank) after https://github.com/lxde/lxqt-session/commit/dc91d5b2225793ea628ceebbef7662b80ef53b27 

![2017-04-25-134223_4160x1080_scrot](https://cloud.githubusercontent.com/assets/159764/25399735/5533b5e6-29be-11e7-854e-24d783380ef5.png)

Qt 5.8.0
Ubuntu 16.04.2 LTS

Errors when run via command line

(non-root)

```bash
 (0x7ffed6ad7650) Warning: dbusCall: QDBusInterface is invalid "org.lxqt.session" "/LXQtSession" "org.lxqt.session" "canLogout"
 (0x7ffed6ad7650) Debug: systemd: "CanPowerOff" = "yes"
 (0x7ffed6ad7650) Debug: systemd: "CanSuspend" = "yes"
 (0x7ffed6ad7650) Debug: systemd: "CanReboot" = "yes"
 (0x7ffed6ad7650) Debug: systemd: "CanHibernate" = "na"
 (0x7ffed6ad7650) Warning: ** Dbus error **************************
 (0x7ffed6ad7650) Warning: Error name  "org.freedesktop.DBus.Error.InvalidArgs"
 (0x7ffed6ad7650) Warning: Error msg   "No such interface 'org.freedesktop.DBus.Properties'"
 (0x7ffed6ad7650) Warning: ****************************************
 (0x7ffed6ad7650) Warning: ** Dbus error **************************
 (0x7ffed6ad7650) Warning: Error name  "org.freedesktop.DBus.Error.AccessDenied"
 (0x7ffed6ad7650) Warning: Error msg   "Rejected send message, 2 matched rules; type=\"method_call\", sender=\":1.1819\" (uid=1000 pid=16857 comm=\"lxqt-leave \") interface=\"org.freedesktop.ConsoleKit.Manager\" member=\"CanHibernate\" error name=\"(unset)\" requested_reply=\"0\" destination=\"org.freedesktop.ConsoleKit\" (uid=0 pid=20193 comm=\"/usr/sbin/console-kit-daemon --no-daemon \")"
 (0x7ffed6ad7650) Warning: ****************************************
 (0x7ffed6ad7650) Debug: systemd: "CanHibernate" = "Rejected send message, 2 matched rules; type=\"method_call\", sender=\":1.1819\" (uid=1000 pid=16857 comm=\"lxqt-leave \") interface=\"org.freedesktop.ConsoleKit.Manager\" member=\"CanHibernate\" error name=\"(unset)\" requested_reply=\"0\" destination=\"org.freedesktop.ConsoleKit\" (uid=0 pid=20193 comm=\"/usr/sbin/console-kit-daemon --no-daemon \")"
```

(root)

```bash
 (0x7ffcf76dfee0) Warning: dbusCall: QDBusInterface is invalid "org.lxqt.session" "/LXQtSession" "org.lxqt.session" "canLogout"
 (0x7ffcf76dfee0) Debug: systemd: "CanPowerOff" = "yes"
 (0x7ffcf76dfee0) Debug: systemd: "CanSuspend" = "yes"
 (0x7ffcf76dfee0) Debug: systemd: "CanReboot" = "yes"
 (0x7ffcf76dfee0) Debug: systemd: "CanHibernate" = "na"
 (0x7ffcf76dfee0) Warning: ** Dbus error **************************
 (0x7ffcf76dfee0) Warning: Error name  "org.freedesktop.DBus.Error.InvalidArgs"
 (0x7ffcf76dfee0) Warning: Error msg   "No such interface 'org.freedesktop.DBus.Properties'"
 (0x7ffcf76dfee0) Warning: ****************************************
 (0x7ffcf76dfee0) Warning: ** Dbus error **************************
 (0x7ffcf76dfee0) Warning: Error name  "org.freedesktop.DBus.Error.UnknownMethod"
 (0x7ffcf76dfee0) Warning: Error msg   "Method \"CanHibernate\" with signature \"\" on interface \"org.freedesktop.ConsoleKit.Manager\" doesn't exist\n"
 (0x7ffcf76dfee0) Warning: ****************************************
 (0x7ffcf76dfee0) Debug: systemd: "CanHibernate" = "Method \"CanHibernate\" with signature \"\" on interface \"org.freedesktop.ConsoleKit.Manager\" doesn't exist\n"
```

**Comment by agaida** [comment]
only a few hints - 16.04 is a LTS, so it is not the right place for brand new and bleeding edge packages. (i don't know the any LXQt packages that are build with Qt 5.8 for 16.04 and i don't know any official Qt 5.8 for 16.04). If one want to play with new things Zesty would fit better - or debian/sid or arch linux.

**Comment by ferronrsmith** [comment]
Not sure I accept that answer. It's still a version of Linux. I can install anything I want on it. A better response would be you don't want to fix the issue. Which is your choice and acceptable.

**Comment by agaida** [comment]
nope - if there would be a bug upstream we would like to know and fix it - it is only very unlikely to get a reproducible bug out of frankensteins xenial - in debian i would have called frankendebian, it's a widely used term.

**Comment by tsujan** [comment]
@ferronrsmith 
We use LXQt, we always want to fix issues and we do fix issues when they're *real*. Here, the problem is that we can't test the latest LXQt on Ubuntu LTS because it's too old for that. Lxqt-leave works OK on our distros, which, as @agaida implicitly said, are up-to-date. LTS distros are good when the user doesn't want frequent updates but they are neither up-to-date *nor good for bleeding-edge packages* (again, as @agaida implied). No developer can do coding under an LTS distro, so there's no guarantee that his *latest* code works there.

**Comment by agaida** [comment]
http://packages.ubuntu.com/xenial/qtbase5-dev
> Package: qtbase5-dev (5.5.1+dfsg-16ubuntu7) 

So we are not in bad mood or not willing - but it is impossible for us to find anything useful

**Comment by palinek** [comment]
I'm really out of any clue, how it can happen nothing is shown.

...but your debug output gave me a hint, that we're not considering the enabled/disabled flag during the key navigation... lxde/lxqt-session@269ed178dc7bc13874aa9dabd730fdb95d981f26

**Comment by ferronrsmith** [comment]
Test on :

Distributor ID:	Ubuntu
Description:	Ubuntu Artful Aardvark (development branch)
Release:	17.10
Codename:	artful

Same issue for me.

**Comment by tsujan** [comment]
> Description:	Ubuntu Artful Aardvark (development branch)

Then, this deserves to be reopened. @palinek, any idea about what could have caused it?

**Comment by tsujan** [comment]
It's interesting that even the Cancel button doesn't have an icon!

@ferronrsmith, do other LXQt apps have icons? Have you installed all LXQt components correctly?

**Comment by agaida** [comment]
tsujan: to make it short - im installing a artful right now :D

**Comment by tsujan** [comment]
@agaida, I wanted to do the same thing on VirtualBox but asked myself what difference could it have from Debian sid regarding LXQt?

**Comment by agaida** [comment]
http://paste.siduction.org/20170426221124.jpg - in ubuntu there is no task-lxqt-desktop :P

**Comment by tsujan** [comment]
So fast! I didn't get what you meant by saying that there was no task-lxqt-desktop in Ubuntu but your screenshot shows Lxqt-leave is OK. Am I right?

**Comment by agaida** [comment]
Steps to reproduce the installation:
* download and install a current mini.iso - pure without anything, just to get a bootable system
* change zesty to artful in sources.list
* apt update; apt dist-upgrade
* apt install lxqt lxqt-common xserver-xorg inxi
done
systemctl start sddm - login and be happy :D

**Comment by palinek** [comment]
Hm...but @ferronrsmith did talk about Qt5.8... on the screenshot in lxqt-about is Qt5.7

**Comment by ferronrsmith** [comment]
it could be that my configuration is some how messed up. My themes changes in pcmanfm-qt doesn't seem to persist (changes on close). 

I build from the source-tree (i like having the latest changes)

**Comment by palinek** [comment]
> I build from the source-tree (i like having the latest changes)

@ferronrsmith just a blind shot... but can you, please, try with this patch (not using homemade item delegate for the view):
```diff
diff --git a/lxqt-leave/listwidget.cpp b/lxqt-leave/listwidget.cpp
index 84826ce..544377a 100644
--- a/lxqt-leave/listwidget.cpp
+++ b/lxqt-leave/listwidget.cpp
@@ -122,11 +122,13 @@ ListWidget::ListWidget(QWidget * parent/* = nullptr*/)
     , mRows(3)
     , mColumns(3)
 {
+    /*
     ItemDelegate * delegate = new ItemDelegate{this};
     {
         QScopedPointer<QAbstractItemDelegate> old_del{itemDelegate()};
         setItemDelegate(delegate);
     }
+    */
 }
 
 void ListWidget::setRows(int rows)
```

**Comment by agaida** [comment]
@palinek - that doesn't matter. there is no Qt 5.8 for debian and ubuntu right now - so i guess that this is a source build with some missed dependencies and a maybe messed up Qt

@ferronrsmith  - and again a few hints: When in Roma do as the romans do. That means, that there is an easy, reliable and reproducible way to get the latest sources and make debian/ubuntu packages.
One should have a look: https://github.com/debian-lxqt/ or the original repos: https://anonscm.debian.org/cgit/pkg-lxqt

**Comment by agaida** [comment]
the branch is lxqt-snapshots - the packaging should be easy to adopt - normally one has to change nothing. one will need devscripts and debhelpers, thats it. Quick and dirty builds can be done with debuild, one could install local packages with apt install ./*deb. One should take the right™ build order, described in wiki.

**Comment by ferronrsmith** [comment]
@palinek

![2017-04-26-183025_4160x1080_scrot](https://cloud.githubusercontent.com/assets/159764/25459902/9dbaf0a4-2aae-11e7-95af-0e9c6edfa647.png)

**Comment by agaida** [comment]
ferronsmith - are the commons installed - is the oxygen icon theme installed? (and selected of course)

**Comment by tsujan** [comment]
I think this should be something in @ferronrsmith's system and we don't need to worry about iconless Lxqt-leave. False positive...

**Comment by agaida** [comment]
and only another hint - it's highly unlikely that Qt 5.8 ever will hit ubuntu or debian. Iirc 5.8 is far to buggy, so the descion in #d-qt-kde was to jump directly to 5.9 - https://tracker.debian.org/pkg/qtbase-opensource-src

**Comment by palinek** [comment]
@ferronrsmith hmm... maybe the `QSize()` c-tor does create a valid `QSize` object with 0,0.... can you, please, test with this patch (and remove the previous one):

```diff
diff --git a/lxqt-leave/listwidget.cpp b/lxqt-leave/listwidget.cpp
index 84826ce..be4d5b9 100644
--- a/lxqt-leave/listwidget.cpp
+++ b/lxqt-leave/listwidget.cpp
@@ -41,6 +41,7 @@
  * \note It is a single purpose delegate and expects, that the model
  * never changes (cached sizeHint() is never invalidated).
  */
+#include <QDebug>
 class ItemDelegate : public QItemDelegate
 {
 public:
@@ -49,6 +50,7 @@ public:
     using QItemDelegate::QItemDelegate;
     virtual QSize sizeHint(const QStyleOptionViewItem &option, const QModelIndex &index) const override
     {
+qDebug() << Q_FUNC_INFO << index << mItemSize;
         if (mItemSize.isValid())
             return mItemSize;
 
```

**Comment by luis-pereira** [comment]
@palinek

> QSize::QSize()
Constructs a size with an invalid width and height (i.e., isValid() returns false).

**Comment by palinek** [comment]
@luis-pereira I'm aware ... but really don't know what can be messed there.

**Comment by tsujan** [comment]
> 5.8 is far to buggy

Actually, the only bug that I've encountered with 5.8 is the focus bug that I explained to Qt devs and was fixed fast (but not in Qt 5.8). However, that bug is very annoying.

> maybe the QSize() c-tor does create a valid QSize object with 0,0

That's impossible, as @luis-pereira implied. This can be closed again, IMO.

> but really don't know what can be messed there.

I don't think anything is messed up there.

**Comment by tsujan** [comment]
In fact, I feel a little guilty for reopening this report and bothering @agaida and @palinek  :(

**Comment by palinek** [comment]
> In fact, I feel a little guilty for reopening this report and bothering @agaida and @palinek :(

No problem here.... actually I'm now really curious

**Comment by ferronrsmith** [comment]
```bash
 (0x7ffd6c667ca0) Warning: dbusCall: QDBusInterface is invalid "org.lxqt.session" "/LXQtSession" "org.lxqt.session" "canLogout"
 (0x7ffd6c667ca0) Debug: systemd: "CanPowerOff" = "yes"
 (0x7ffd6c667ca0) Debug: systemd: "CanSuspend" = "yes"
 (0x7ffd6c667ca0) Debug: systemd: "CanReboot" = "yes"
 (0x7ffd6c667ca0) Debug: systemd: "CanHibernate" = "na"
 (0x7ffd6c667ca0) Warning: ** Dbus error **************************
 (0x7ffd6c667ca0) Warning: Error name  "org.freedesktop.DBus.Error.InvalidArgs"
 (0x7ffd6c667ca0) Warning: Error msg   "No such interface 'org.freedesktop.DBus.Properties'"
 (0x7ffd6c667ca0) Warning: ****************************************
 (0x7ffd6c667ca0) Warning: ** Dbus error **************************
 (0x7ffd6c667ca0) Warning: Error name  "org.freedesktop.DBus.Error.AccessDenied"
 (0x7ffd6c667ca0) Warning: Error msg   "Rejected send message, 2 matched rules; type=\"method_call\", sender=\":1.5118\" (uid=1000 pid=10676 comm=\"lxqt-leave \") interface=\"org.freedesktop.ConsoleKit.Manager\" member=\"CanHibernate\" error name=\"(unset)\" requested_reply=\"0\" destination=\"org.freedesktop.ConsoleKit\" (uid=0 pid=24866 comm=\"/usr/sbin/console-kit-daemon --no-daemon \")"
 (0x7ffd6c667ca0) Warning: ****************************************
 (0x7ffd6c667ca0) Debug: systemd: "CanHibernate" = "Rejected send message, 2 matched rules; type=\"method_call\", sender=\":1.5118\" (uid=1000 pid=10676 comm=\"lxqt-leave \") interface=\"org.freedesktop.ConsoleKit.Manager\" member=\"CanHibernate\" error name=\"(unset)\" requested_reply=\"0\" destination=\"org.freedesktop.ConsoleKit\" (uid=0 pid=24866 comm=\"/usr/sbin/console-kit-daemon --no-daemon \")"
 (0x7ffd6c667ca0) Debug: virtual QSize ItemDelegate::sizeHint(const QStyleOptionViewItem&, const QModelIndex&) const QModelIndex(0,0,0x55b0b0c03850,QListModel(0x55b0b0b3ba00)) QSize(-1, -1)
 (0x7ffd6c667ca0) Debug: virtual QSize ItemDelegate::sizeHint(const QStyleOptionViewItem&, const QModelIndex&) const QModelIndex(5,0,0x55b0b0c03530,QListModel(0x55b0b0b3ba00)) QSize(96, 24)
 (0x7ffd6c667ca0) Debug: virtual QSize ItemDelegate::sizeHint(const QStyleOptionViewItem&, const QModelIndex&) const QModelIndex(0,0,0x55b0b0c03850,QListModel(0x55b0b0b3ba00)) QSize(96, 24)


You can close this. I'll use the patch up above.
```

**Comment by agaida** [comment]
tsujan: no problem here too - but as you said the important changes and fixes will be in 5.9, like fixed platform things and many others.

**Comment by tsujan** [comment]
@agaida I've been waiting for 5.9 for months, having patched 5.8 to get rid of its focus bug with every Manjaro update. 5.8 was a mistake.

**Comment by agaida** [comment]
i guess that once stretch is released the floodgates of hell will be opened - and there will be an Qt 5.9 really fast in sid. Until then i tend to ignore it as far as i can - 5.9 will mean a lot of symbol changes in my packages. And that will be really pita. Beside of that i want to have 5.9 as soon as possible :P

**Comment by palinek** [comment]
I've reproduced it.... our item delegate has some problem when there is no icon... I'll investigate...

**Comment by palinek** [comment]
Heh... the problem is deeply in one of these (each will solve the problem):
1. the discrepancy in `QItemDelegate::sizeHint()` (and calling the `QItemDelegate::rect()` which doesn't call `QIcon::actualSize()` on *null* icon) and `QItemDelegate::paint()` (and calling the `QIcon::actualSize()` on *null* icon)
2. the implementation in libqtxdg/xdgiconloader in `XdgIconLoaderEngine::actualSize()`, where if no suitable entry found the default `QIconEnginge::actualSize()` is returned (btw. the default Qt's `QIconLoaderEngine` suffers by this too)
3. make the default implementation `QIconEngine::actualSize()` return the empty `QSize{0, 0}`

IMO the most straight-forward is to fix by 2. (and optionally submit the fix also to Qt).

**Comment by palinek** [comment]
@ferronrsmith Can you, please, try with the lxde/libqtxdg#130?

**Comment by ferronrsmith** [comment]
doing that now.

**Comment by palinek** [comment]
> doing that now. Will let you know the outcome

Actually... are you using the lxqt-qt-plugin? If not, the change will not have any effect for you.

**Comment by ferronrsmith** [comment]
Yea you are right, I am using this with `i3`, i think that might have messed up some things along the way

**Comment by palinek** [comment]
> Yea you are right, I am using this with i3

It doesn't contradict.

**Comment by ferronrsmith** [comment]
think I might have not configured it properly to begin with, because nothing works as it should.I can live without this.

**Comment by palinek** [comment]
FYI ... submitted also to Qt -> https://codereview.qt-project.org/#/c/192904/

**Comment by ferronrsmith** [comment]
I tested on the 5.9 beta 2 same error :|

**Comment by palinek** [comment]
> I tested on the 5.9 beta 2 same error :|

Try to apply the patch from codereview manually.

**Comment by mageddo** [comment]
It's also happening on my Lubuntu 18.10, is the source of the problem the same?
```
lxqt-leave --hibernate

 (0x7ffd4d782c90) Warning: Icon theme "breeze" not found.
 (0x7ffd4d782c90) Debug: systemd: "CanHibernate" = "no"
 (0x7ffd4d782c90) Warning: ** Dbus error **************************
 (0x7ffd4d782c90) Warning: Error name  "org.freedesktop.DBus.Error.InvalidArgs"
 (0x7ffd4d782c90) Warning: Error msg   "No such interface “org.freedesktop.DBus.Properties”"
 (0x7ffd4d782c90) Warning: ****************************************
 (0x7ffd4d782c90) Warning: dbusCall: QDBusInterface is invalid "org.freedesktop.ConsoleKit" "/org/freedesktop/ConsoleKit/Manager" "org.freedesktop.ConsoleKit.Manager" "CanHibernate"
```

**Comment by agaida** [comment]
Without knowing much about Lubuntu - may it be that you changed some basic things like the init system?

**Comment by tsimonq2** [comment]
No, the errors are not related.

**Comment by tumpio** [comment]
@mageddo this was a permission issue. I fixed it by following this tutorial which includes rule for Polkit:

https://fitzcarraldoblog.wordpress.com/2018/07/14/configuring-lubuntu-18-04-to-enable-hibernation-using-a-swap-file/

> 7.2  For Polkit versions below 0.106

`sudo nano /var/lib/polkit-1/localauthority/50-local.d/50-enable-suspend-on-lockscreen.pkla`

```
[Allow hibernation and suspending with lock screen]
Identity=unix-user:*
Action=org.freedesktop.login1.suspend;org.freedesktop.login1.suspend-multiple-sessions;org.freedesktop.login1.hibernate;org.freedesktop.login1.hibernate-multiple-sessions
ResultAny=yes
ResultInactive=yes
ResultActive=yes
```

Before adding this rule, the hibernate entry was disabled in the "Leave" menu. Now it is enabled and working. Should this Polkit rule be included by default in Lubuntu?

**Comment by agaida** [comment]
thats a perfect downstream question

**Comment by cxw42** [comment]
@tumpio Thank you!  The same fix worked for me on Lubuntu 19.10.  I'm not sure how to report this downstream --- could anyone give me any pointers?  Much appreciated!
